### PR TITLE
Create workflow trigger for publishing release image

### PR DESCRIPTION
**Pull request by eradman**
## What type of PR is this? 

- [x] Other

## Description

- If the action was not triggered manually, publish a preview image
- Always select a branch for release image

## How is this tested?

- [x] Manually

Tested on personal Dockerhub account

https://hub.docker.com/repository/docker/eradman/redash/general
https://hub.docker.com/repository/docker/eradman/preview/general

## Related Tickets & Documents

https://github.com/getredash/redash/issues/7251

Preview image from master

preview-image-1

Publish a release

preview-image-2

### CODE REVIEW

File: .github/workflows/preview-image.yml [hunk 1]
```diff
@@ -75,11 +84,28 @@ jobs:
...
       - name: Build and push preview image to Docker Hub
         uses: docker/build-push-action@v4
+        if: ${{ github.event.inputs.dockerRepository == 'preview' || !github.event.workflow_run }}
         with:
           push: true
           tags: |
-            redash/redash:preview
-            redash/preview:${{ steps.version.outputs.VERSION_TAG }}
+            eradman/redash:preview
+            eradman/preview:${{ steps.version.outputs.VERSION_TAG }}
```
Comment (lucydodo) [inline .github/workflows/preview-image.yml:92]:
```suggestion
            redash/redash:preview
            redash/preview:${{ steps.version.outputs.VERSION_TAG }}
```

File: .github/workflows/preview-image.yml [hunk 2]
```diff
@@ -75,11 +84,28 @@ jobs:
...
+        env:
+          DOCKER_CONTENT_TRUST: true
+
+      - name: Build and push release image to Docker Hub
+        uses: docker/build-push-action@v4
+        if: ${{ github.event.inputs.dockerRepository == 'redash' }}
+        with:
+          push: true
+          tags: |
+            eradman/redash:${{ steps.version.outputs.VERSION_TAG }}
```
Comment (lucydodo) [inline .github/workflows/preview-image.yml:108]:
```suggestion
            redash/redash:${{ steps.version.outputs.VERSION_TAG }}
```

**Review by lucydodo — COMMENTED** [review]
It looks good, but it seems like you might have forgotten to change the username back after testing. 🤔

File: .github/workflows/preview-image.yml (hunk 1 above)
Comment (eradman) [reply .github/workflows/preview-image.yml:92]: Ah, good catch

**Review by justinclift — APPROVED** [review]
With that repo name fixed, this looks good to merge. :smile:

**Pull request merged**